File: exercism/go/archive/difference-of-squares/difference_of_squares.go
// Package diffsquares gives us the difference between the sum of squares vs the squared sum of
// the first N natural numbers
package diffsquares

// SquareOfSums returns the squared value of the sum of the first N natural numbers
func SquareOfSums(num int) (total int) {
	if num <= 0 {
		return 0
	}

	for i := 1; i <= num; i++ {
		total += i
	}

	return total * total
}

// SumOfSquares returns the sum of the squares of the first N natural numbers
func SumOfSquares(num int) (total int) {
	if num <= 0 {
		return 0
	}

	for i := 1; i <= num; i++ {
		total += i * i
	}

	return total
}

// Difference yields the gap between two numbers
func Difference(num int) (diff int) {
	if num <= 0 {
		return 0
	}

	return SquareOfSums(num) - SumOfSquares(num)
}

File: exercism/python/archive/difference-of-squares/difference_of_squares.py
""" Module to perform permutations of the difference
of squares of the first N natural numbers """
# -*- coding: utf-8 -*-

from __future__ import unicode_literals

import math


def square_of_sum(num):
    """ return the square of the sum of the first N natural numbers """
    return math.pow(sum([a for a in range(1, num+1) if a]), 2)


def sum_of_squares(num):
    """ return the sum of the squares of the first N natural numbers """
    return sum([math.pow(a, 2) for a in range(1, num+1) if a])


def difference(num):
    """ return the difference between the square_of_sum and sum_of_squares
    for the first N natural numbers
    """
    return square_of_sum(num) - sum_of_squares(num)
